rest/routing.go

```diff
@@ -345,6 +345,12 @@ func CreateMetricHandler(sc *ServerContext) http.Handler {
 	return wrapRouter(sc, metricsPrivs, router)
 }
 
+// CreateDiagnosticHandler Creates the HTTP handler for the diagnostic API of a gateway server.
+func CreateDiagnosticHandler(sc *ServerContext) http.Handler {
+	router := CreateDiagnosticRouter(sc)
+	return wrapRouter(sc, diagnosticPrivs, router)
```

Comment:
Let's drop `diagnosticPrivs` for now, and just use `adminPrivs` until it is necessary.